### ISSUE TITLE
Widget buttons don't work when using the New Unity Input System

### ISSUE THREAD

**Issue by SlavomirDev** [googlevr/cardboard#418]
Tracking clicks on Close and Gear buttons does not work when using the new Unity input system

- Any devices
- Any OS

How to reproduce:
1. Just click on the widget buttons

### VERSIONS USED
Cardboard SDK/Cardboard XR Plugin 1.21.0

### UNITY VERSIONS USED
- Unity version: 2021.3.2f1

![photo_2023-06-15_10-08-29](https://github.com/googlevr/cardboard/assets/135220037/123e69a5-2c0c-455a-9c9a-b6ce154a7eb8)

Properties Api.IsCloseButtonPressed and Api.IsGearButtonPressed always as false if you are using New Unity Input System. Input.GetTouch() - does not work, it is not possible to track the position of touch through the old Input class
I already have a ready-made solution for this:
```c#
        public static bool IsCloseButtonPressed
        {
            get
            {
                if (!XRLoader._isStarted)
                    return false;
#if ENABLE_INPUT_SYSTEM
                Touchscreen touchscreen = Touchscreen.current;

                if (touchscreen == null || !touchscreen.added)
                    return false;

                var touches = touchscreen.touches;
                if (touches.Count <= 0)
                    return false;

                var touch = touches[0];
                if (touch == null || !touch.IsPressed() || !touch.isInProgress)
                    return false;

                Vector2Int touchPosition = Vector2Int.RoundToInt(touch.position.ReadValue());
                return Widget.CloseButtonRect.Contains(touchPosition);
#else

                if (Input.touchCount == 0)
                    return false;

                Touch touch = Input.GetTouch(0);
                Vector2Int touchPosition = Vector2Int.RoundToInt(touch.position);
                return touch.phase == TouchPhase.Began
                    && Widget.CloseButtonRect.Contains(touchPosition);
#endif
            }
        }

        public static bool IsGearButtonPressed
        {
            get
            {
                if (!XRLoader._isStarted)
                    return false;

#if ENABLE_INPUT_SYSTEM
                Touchscreen touchscreen = Touchscreen.current;

                if (touchscreen == null || !touchscreen.added)
                    return false;

                var touches = touchscreen.touches;
                if (touches.Count <= 0)
                    return false;

                var touch = touches[0];
                if (touch == null || !touch.IsPressed() || !touch.isInProgress)
                    return false;

                Vector2Int touchPosition = Vector2Int.RoundToInt(touch.position.ReadValue());
                return Widget.GearButtonRect.Contains(touchPosition);
#else

                if (Input.touchCount == 0)
                    return false;

                Touch touch = Input.GetTouch(0);
                Vector2Int touchPosition = Vector2Int.RoundToInt(touch.position);
                return touch.phase == TouchPhase.Began
                    && Widget.GearButtonRect.Contains(touchPosition);
#endif
            }
        }
``` 
It would be great if you adapted your SDK to the new Input System

**Comment by arilow** [comment]
Thanks @SlavomirDev for reaching out to us!

PR [#24](https://github.com/googlevr/cardboard-xr-plugin/pull/24) proposes a similar solution. An internal ticket has been already created to address this feature request. We'll update this ticket once we have some news about it. 

Thanks!

**Comment by arilow** [comment]
Closing as this was fixed in v1.25. In case this is still an issue for you, please feel free to reopen this ticket. Thanks!
